### ISSUE TITLE
Data.json Harvester bug fixes

### ISSUE THREAD

**Issue by kimwdavidson** [GSA/datagov-ckan-multi#193]
In the legacy data.json extension, we have two validators for federal and non-federal data.json sources.

If you don't configure the harvest source, by default, will be treated as federal.

If it's a federal one, bureauCode is required

The city of Boise has its config correctly defined.

The new harvester is missing this step. We treat all sources the same. We need to fix that and use this JSON validation files.

Tasks

 - [x] Read the havest source configuration in order to detect if the source is federal or not

 - [x]  Validate the data depending on this configuration. Use the JSON validation files

 - [x]  Test this functionality

**Comment by avdata99** [comment]
This was managed through these issues:
 - At the harvester core library: https://gitlab.com/datopian/ckan-ng-harvester-core/issues/1
 - At the harvester app: https://gitlab.com/datopian/ckan-ng-harvest/issues/64

These changes are waiting for approval at this MR: https://gitlab.com/datopian/ckan-ng-harvest/merge_requests/70

**Comment by kimwdavidson** [comment]
Blocked pending cluster fixes

**Comment by avdata99** [comment]
We deploy and detect an error: https://gitlab.com/datopian/ckan-ng-harvest/issues/68

**Comment by estebanruseler** [comment]
Fixed this issue, whilst fixing we found other small bugs. Created a new issue for those.

**Comment by kimwdavidson** [comment]
@avdata99 merging, testing locally. adam to deploy to sandbox, then move to QA

**Comment by akariv** [comment]
@avdata99 please comment here with the latest ckan-ng-harvest docker image tag so I can redeploy when ready.

**Comment by avdata99** [comment]
Uploading to [avdata99/ckan-ng-harvest](https://hub.docker.com/repository/docker/avdata99/ckan-ng-harvest) @akariv. Will be upload in the next 2 hours.

**Comment by akariv** [comment]
@avdata99 Working with manually built images is ill-advised.
ckan-ng-harvest docker images should be automatically built in CI and get uploaded here: https://hub.docker.com/r/viderum/ckan-ng-harvest

This was done by GitLab CI - when moving from GitLab CI to Travis CI, this functionality needs to also be migrated.

**Comment by avdata99** [comment]
We expect to build and upload docker images at #8 (https://github.com/datopian/ckan-ng-harvest/issues/8).
The issue was fixed. We can do some QA just locally by now.
I also built and upload as avdata99/ckan-ng-harvest to test.
I think this is not related to the issue and we can close this.

**Comment by akariv** [comment]
To test locally it's fine, but in order to deploy to the sandbox it needs to be properly built.

**Comment by avdata99** [comment]
Done [here](https://github.com/datopian/ckan-ng-harvest/issues/8#issuecomment-569333314)

**Comment by akariv** [comment]
Please update the harvester docker image tag in the
datagov-ckan-provisioning-api PR you opened as 'harvesterImage'

**Comment by kimwdavidson** [comment]
Duplication error ready for QA

**Comment by adborden** [comment]
If I'm understanding the history correctly, https://github.com/datopian/ckan-ng-harvest is now building docker images to be consumed by https://github.com/ViderumGlobal/datagov-ckan-cloud-provisioning-api. The data.json bugfixes described in the issue are done and published in the latest ckan-ng-harvest image, but we still need to update https://github.com/ViderumGlobal/datagov-ckan-cloud-provisioning-api/pull/3 before those fixes are in an environment where we can QA.

I'm going to move this back to `In Progress` until https://github.com/ViderumGlobal/datagov-ckan-cloud-provisioning-api/pull/3 is merged. Please holler if that's not the case.

**Comment by avdata99** [comment]
Ok. @akariv could merge this and update the sandox to test it.

**Comment by estebanruseler** [comment]
@akariv this is now unblocked so moving this in progress. @avdata99 has tested locally and in sandbox. Did you want to do additional testing or can we move to QA.

**Comment by estebanruseler** [comment]
@moved to QA. This can be tested along with 195. I can't seem to re-assign to Aaron

**Comment by adborden** [comment]
:+1:
